independent-projects/arc/runtime/src/main/java/io/quarkus/arc/BindingsSource.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.arc;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Defines a source of interception binding annotations. When this annotation
+ * is present on an {@link InterceptionProxy} parameter of a CDI producer method,
+ * the interception binding annotations present on the target class are ignored
+ * and instead are read from the class defined by this annotation.
+ * <p>
+ * Interceptor bindings for method named {@code M} with return type {@code R}
+ * and parameter types {@code P1, ..., Pn} on the target class are obtained from
+ * a method with an identical name, return type and parameter types declared
+ * on the bindings source class.
+ * <p>
+ * The annotations that can be meaningfully present on the {@code value} class are:
+ * <p>
+ * <ul>
+ * <li><em>interceptor bindings</em>: on the class and on the methods</li>
+ * <li><em>stereotypes</em>: on the class</li>
+ * <li>{@link NoClassInterceptors}: on the methods</li>
+ * </ul>
+ * <p>
+ * Other annotations on the {@code value} class are ignored.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.PARAMETER)
+public @interface BindingsSource {
+    /**
+     * The class from which interceptor binding annotations should be read.
+     */
+    Class<?> value();
```

Comment:
I wonder if it would make sense to support multiple binding sources? I.e. either make this annotation repeatable or use  `Class<?>[] value()`...

Reply:
I'm have no idea how combining multiple bindings sources would work. I would stay away from that if possible.

Reply:
I was thinking about simple merge but you're right that for interceptor bindings with members we would need priorities and such...

Reply:
You can do that ofc but my original idea was about sharing the bindings, e.g. when you declare multiple bindings sources for a class hierarchy.

Reply:
I understand, I just think that might be an overkill, at least initially, as it adds complexity to the implementation with little usage/upside.

If we want to share them, then we could instead allow the single class declared here to have subclasses that we also inspect. That could follow rules very similar to what interceptors already do and you wouldn't run into conflicts with priorities, annotation member values etc. That being said, I still vote for the current design :)